breadth_first_search/BreadhFirstSearch.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <list>
+using namespace std;
+
+class Graph {
+	int v;
+	list<int> *adjlist;
+	public :
+
+	Graph(int V) {
+		v = V;
+		adjlist = new list<int>[V];
+	}
+
+	void addEdge (int source, int dest) {
+		adjlist[source].push_back(dest);
+	}
+
+	void breadthFirstSearch (int source) {
+		bool *visited = new bool[v];
+		list<int> queue;
+
+		visited[source] = true;
+		queue.push_back(source);
+
+		list<int>::iterator itr;
+		while (!queue.empty()) {
+			source = queue.front();
+			queue.pop_front();
+			cout<<source<<"->";
+
+			for(itr = adjlist[source].begin(); itr != adjlist[source].end(); itr++) {
```

Comment:
Try to use `++itr` instead of `itr++`. For efficiency reason.[Refer](https://stackoverflow.com/questions/1077026/incrementing-iterators-it-more-efficient-than-it)